lib/trie/trie.go

```diff
@@ -322,6 +342,9 @@ func findNextKeyChild(children []*Node, startIndex byte,
 // Put inserts a value into the trie at the
 // key specified in little Endian format.
 func (t *Trie) Put(keyLE, value []byte) {
+	defer func() {
+		t.handleTrackedDeltas(nil)
+	}()
```

Comment:
is there any reason to put `t.handleTrackedDeltas(nil)` inside a defer call?
```suggestion
	nibblesKey := codec.KeyLEToNibbles(keyLE)
	t.root, _, _ = t.insert(t.root, nibblesKey, value)
        t.handleTrackedDeltas(nil)
```

Reply:
Well not for now, but there is on the next branch where `nil` gets replaced with `err` because `insert` returns an error.
If you don't mind can we keep it as is (especially so I don't forget to put it back in a defer call).